src/Esfa.Vacancy.Register.Infrastructure/Repositories/VacancyRepository.cs

```diff
@@ -1,13 +1,71 @@
-﻿using System.Threading.Tasks;
+﻿using Dapper;
+using Esfa.Vacancy.Register.Domain.Repositories;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using DomainEntities = Esfa.Vacancy.Register.Domain.Entities;
 
 namespace Esfa.Vacancy.Register.Infrastructure.Repositories
 {
     public class VacancyRepository : IVacancyRepository
     {
-        public async Task<Domain.Entities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
+        private readonly IProvideSettings _provideSettings;
+
+        public VacancyRepository(IProvideSettings provideSettings)
+        {
+            _provideSettings = provideSettings;
+        }
+
+        public async Task<DomainEntities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
         {
-            await Task.Delay(10);
-            return null;
+            var connectionString = _provideSettings.GetSetting(ApplicationSettingConstants.AvmsPlusDatabaseConnectionStringKey);
+
+            DomainEntities.Vacancy vacancy;
+
+            using (var sqlConn = new SqlConnection(connectionString))
+            {
+                var query = $"{GetVacancyDetailsQuery} AND V.VacancyReferenceNumber = {referenceNumber}";
```

Comment:
I know that referenceNumber is an int, but it would be less scary to see it parameterised for the next person think of SQL injection.